File: Autolock/Bluelink/BluelinkSessionStore.swift
import Foundation

/// Persists the Bluelink session (tokens + vehicle info) separately from
/// CredentialsStore's account/password/PIN, so clearing a stale session
/// never requires re-entering credentials.
enum BluelinkSessionStore {
    private static let account = "bluelink-session"

    static func save(_ session: BluelinkSession) throws {
        try KeychainStore.save(session, account: account)
    }

    static func load() throws -> BluelinkSession? {
        try KeychainStore.load(BluelinkSession.self, account: account)
    }

    static func clear() throws {
        try KeychainStore.delete(account: account)
    }
}

File: android/app/src/main/java/com/sendprobe/autolock/storage/BluelinkSessionStore.kt
package com.sendprobe.autolock.storage

import android.content.Context
import com.sendprobe.autolock.bluelink.BluelinkSession

/**
 * Persists the Bluelink session (tokens + vehicle info) separately from
 * CredentialsStore's account/password/PIN, so clearing a stale session
 * never requires re-entering credentials.
 */
object BluelinkSessionStore {
    private const val ACCOUNT = "bluelink-session"

    fun save(context: Context, session: BluelinkSession) {
        SecureStore.save(context, session, ACCOUNT)
    }

    fun load(context: Context): BluelinkSession? =
        SecureStore.load(context, ACCOUNT)

    fun clear(context: Context) {
        SecureStore.delete(context, ACCOUNT)
    }
}
